cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -311,18 +339,26 @@ func (scaleSet *ScaleSet) IncreaseSize(delta int) error {
 		return fmt.Errorf("size increase too large - desired:%d max:%d", int(size)+delta, scaleSet.MaxSize())
 	}
 
-	return scaleSet.SetScaleSetSize(size + int64(delta))
+	return scaleSet.setScaleSetSize(size+int64(delta), delta)
 }
 
 // GetScaleSetVms returns list of nodes for the given scale set.
 func (scaleSet *ScaleSet) GetScaleSetVms() ([]compute.VirtualMachineScaleSetVM, *retry.Error) {
-	klog.V(4).Infof("GetScaleSetVms: starts")
 	ctx, cancel := getContextWithTimeout(vmssContextTimeout)
 	defer cancel()
 
-	resourceGroup := scaleSet.manager.config.ResourceGroup
-	vmList, rerr := scaleSet.manager.azClient.virtualMachineScaleSetVMsClient.List(ctx, resourceGroup, scaleSet.Name, "instanceView")
+	var vmList []compute.VirtualMachineScaleSetVM
+	var rerr *retry.Error
+	if scaleSet.enableDetailedCSEMessage {
+		vmList, rerr = scaleSet.manager.azClient.virtualMachineScaleSetVMsClient.List(ctx, scaleSet.manager.config.ResourceGroup,
+			scaleSet.Name, string(compute.InstanceViewTypesInstanceView))
+	} else {
+		vmList, rerr = scaleSet.manager.azClient.virtualMachineScaleSetVMsClient.List(ctx, scaleSet.manager.config.ResourceGroup,
```

Comment:
as long as it is turn on by default, this won't be triggered

Reply:
In fact it won't be turned on by default... Will probably revert this back for now, given provisioning state handling differences rely on instance view.